components/button/src/Button.js

```diff
@@ -0,0 +1,276 @@
+		height: button.size[size].height,
+	};
+};
+
+const getBlockStyling = block => {
+	return block
+		? // Block mode
+		  {
+				display: 'flex', //flex will provide 'block-level' appearance (we use flex so any icon children can be positioned appropriately)
+				width: '100%',
+		  }
+		: // Inline mode
+		  {
+				display: 'inline-flex',
+				width: 'auto', //reset
+		  };
+};
+
+// ==============================
+// Component
+// ==============================
+
+export const Button = ({
+	appearance,
+	size,
+	soft,
+	block,
+	trim,
+	icon: Icon,
+	iconPosition,
+	justify,
+	children,
+	tag: Tag,
+	onClick,
+	...props
+}) => {
+	const theme = useTheme();
+	const mq = mediaQueries(theme.breakpoints);
+	const button = theme.button;
+
+	// Common styling
+	const styleCommon = {
+		alignItems: 'center', //vertical
+		appearance: 'none',
+		border: `${button.borderWidth} solid transparent`,
+		borderRadius: button.borderRadius,
+		cursor: 'pointer',
+		flex: 1,
+		flexDirection: iconPosition === 'left' ? 'row-reverse' : null,
+		fontWeight: button.fontWeight,
+		justifyContent: justify ? 'space-between' : 'center', //horizontal
+		lineHeight: button.lineHeight,
+		textAlign: 'center',
+		textDecoration: 'none',
+		touchAction: 'manipulation',
+		transition: 'background 0.2s ease, color 0.2s ease',
+		userSelect: 'none',
+		verticalAlign: 'middle',
+		whiteSpace: 'nowrap',
+
+		// Hover state (but excluded if disabled or inside a disabled fieldset)
+		':hover:not(:disabled), fieldset:not(:disabled) &:hover': {
+			textDecoration: appearance === 'link' ? 'underline' : 'none',
+		},
+
+		// Disabled via `disabled` attribute or inside a disabled fieldset
+		':disabled, fieldset:disabled &': {
+			opacity: '0.5',
+			pointerEvents: 'none',
+		},
+	};
+
+	// Button appearance styling
+	const styleAppearance = {
+		...button.appearance[appearance].standard.default,
+		...(soft && appearance !== 'link' && button.appearance[appearance].soft.default),
+
+		':hover': {
+			...button.appearance[appearance].standard.hover,
+			...(soft && appearance !== 'link' && button.appearance[appearance].soft.hover),
+		},
+		':active, &.active': {
+			...button.appearance[appearance].standard.active,
+			...(soft && appearance !== 'link' && button.appearance[appearance].soft.active),
+		},
+	};
+
+	// Button size styling
+	const styleSize = Array.isArray(size)
+		? // Size provided as an array
+		  size.map((s, i) => {
+				return i === 0
+					? getSizeStyling(s, trim, button)
+					: { [mq[i]]: getSizeStyling(s, trim, button) };
+		  })
+		: // Size prop provided a string, returned as single index array
+		  [getSizeStyling(size, trim, button)];
+
+	// Block styling
+	const styleBlock = Array.isArray(block)
+		? // Block provided as an array
+		  block.map((b, i) => {
+				return i === 0 ? getBlockStyling(b) : { [mq[i]]: getBlockStyling(b) };
+		  })
+		: // Block prop provided a string, returned as single index array
+		  [getBlockStyling(block)];
+
```

Comment:
Refactoring to use facepaint we can replace styleSize, getSizeStyling(), styleBlock, getBlockStyling() to
```
// new util
const asArray = val => (Array.isArray(val) ? val : [val]);

const styleResponsive = () => {
		const sizeArr = asArray(size);
		const blockArr = asArray(block);

		const padding = [],
			fontSize = [],
			height = [];

		sizeArr.forEach(s => {
			const pad = button.size[s].padding;
			if (trim) pad[1] = 0;
			padding.push(pad.join(' '));
			fontSize.push(button.size[s].fontSize);
			height.push(button.size[s].height);
		});

		return {
			padding,
			fontSize,
			height,
			display: blockArr.map(b => (b ? 'flex' : 'inline-flex')),
			width: blockArr.map(b => (b ? '100%' : 'auto')),
		};
	};
```